pkg/controllers/provisioning/scheduling/events/events.go

```diff
@@ -20,23 +20,39 @@ import (
 	v1 "k8s.io/api/core/v1"
 	"k8s.io/client-go/util/flowcontrol"
 
+	"github.com/aws/karpenter-core/pkg/apis/v1alpha5"
 	"github.com/aws/karpenter-core/pkg/events"
 )
 
 // PodNominationRateLimiter is a pointer so it rate-limits across events
 var PodNominationRateLimiter = flowcontrol.NewTokenBucketRateLimiter(5, 10)
 
-func NominatePod(pod *v1.Pod, node *v1.Node) []events.Event {
-	return []events.Event{
-		{
+// PodNominationRateLimiterForMachine is a pointer so it rate-limits across events
+var PodNominationRateLimiterForMachine = flowcontrol.NewTokenBucketRateLimiter(5, 10)
+
+func NominatePod(pod *v1.Pod, node *v1.Node, machine *v1alpha5.Machine) []events.Event {
+	var evts []events.Event
+	if node != nil {
```

Comment:
Building on previous comment, it feels odd to me that we'd emit an event for a machine, but not its node. Maybe this is not something to worry about.

Reply:
We event for both here exactly for that reason. When we first schedule, we don't know the node so we inherently have to ignore the node event